### PR TITLE
[PATCH v3] validation: crypto: clean up code a bit

### PR DESCRIPTION
**Pull request by JannePeltonen**
validation: crypto: add crypto algorithm IDs in test vectors
validation: crypto: simplify crypto_op() function
validation: crypto: move code to separate files

### CODE REVIEW

**Comment by JannePeltonen** [comment]
v2: rebased

**Review by JereLeppanen — APPROVED** [review]

File: test/validation/api/crypto/test_vectors.h [hunk 1]
```diff
@@ -8,10 +8,13 @@
 #ifndef _ODP_TEST_CRYPTO_VECTORS_H_
```
Comment (TuomasTaipale) [inline test/validation/api/crypto/test_vectors.h]: Copyright could be updated. There are also some dead entries in AES XTS section that have been commented out for a few years, maybe those could be just removed.

Comment (JannePeltonen) [reply test/validation/api/crypto/test_vectors.h]: I do not think the copyright year should be updated when there are no changes other than just the moving of code around.

The commented-out XTS vector could indeed be removed or uncommented but that is better done in a separately. So I am not including it in this PR.

Comment (TuomasTaipale) [reply test/validation/api/crypto/test_vectors.h]: You added two new fields, not just moving code around.

Comment (JannePeltonen) [reply test/validation/api/crypto/test_vectors.h]: ok, I'll update the copyright.

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**